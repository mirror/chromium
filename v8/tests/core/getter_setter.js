function check_getter_setter(obj) {
  var sum = 0;
  for(var name in obj) sum += obj[name];

  print(obj.fisk);
  print(obj["kalle"]);
  print(sum);
  obj.kalle = 4;
  
  print(typeof obj.__lookupGetter__("kalle") == 'function');
  print(typeof obj.__lookupSetter__("kalle") == 'function');
}

// Check setter and getter defined by primitives.
var a = { fisk : 2 }
a.__defineGetter__("kalle", function() { return this.fisk * 2; });
a.__defineSetter__("kalle", function(value) { print(this.fisk * value); });
check_getter_setter(a);

// Check setter and getter defined by object literal.
check_getter_setter({
  get kalle() { return this.fisk * 2; },
  set kalle(value) { print(this.fisk * value); },
  fisk: 2
})

// Verify getter can overwrite a normal property.
var o = { x: 'old x', y: 'old y' };
o.__defineGetter__('x', function() { return 'new x'; });
print(o.x);

// Verify setter can overwrite a normal property.
o.__defineSetter__('y', function(value) { this.y_mirror = value; });
o.y = 'new y';
print(o.y_mirror);
