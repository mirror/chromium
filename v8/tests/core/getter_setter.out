2
4
6
8
true
true
2
4
6
8
true
true
new x
new y
